tests/Model/IndexInfoFunctionalTest.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+namespace MongoDB\Tests\Model;
+
+use MongoDB\Collection;
+use MongoDB\Operation\CreateIndexes;
+use MongoDB\Operation\ListIndexes;
+use MongoDB\Tests\Operation\FunctionalTestCase;
+
+class IndexInfoFunctionalTest extends FunctionalTestCase
+{
+    private $collection;
+
+    public function setUp()
+    {
+        parent::setUp();
+
+        $this->collection = new Collection($this->manager, $this->getDatabaseName(), $this->getCollectionName());
+    }
+
+    public function testIs2dSphere()
+    {
+        $this->collection->createIndex(['pos' => '2dsphere']);
+        $result = $this->collection->listIndexes();
+
+        $result->rewind();
+        $this->assertInstanceOf('MongoDB\Model\IndexInfo', $result->current());
+        $result->next();
+        $index = $result->current();
+
+        $this->assertTrue($index->is2dSphere());
+
+        if (version_compare($this->getServerVersion(), '3.2.0', '<')) {
+            $this->assertEquals(2, $index['2dsphereIndexVersion']);
+            return;
```

Comment:
Consider:

```
$expectedVersion = version_compare($this->getServerVersion(), '3.2.0', '<') ? 2 : 3;
$this->assertEquals($expectedVersion, $index['2dsphereIndexVersion']);
```

This ensures that if more assertions are later appended to this test, we limit the chances of a bug missing these for < 3.2.